eclipse-theia/theia
# core: pass `context` to when clause for toolbar items

#### What it does


Fixes: #12314 
Fixes: https://github.com/eclipse-theia/theia/pull/12295#issuecomment-1472767481

The pull-request fixes an issue when evaluating when clauses for toolbar items which were not aware of the current context making the toolbar items never enabled or executable. The change passes the context when it exists to ensure we properly apply the clause.

#### How to test


1. confirm the steps from #12067 pass
2. confirm with [vscode-trace-extension-0.1.0.vsix..zip](https://github.com/eclipse-theia/theia/files/11030608/vscode-trace-extension-0.1.0.vsix.zip) that the trace-viewer view has a toolbar item for "open traces" and that it works well
3. bonus: confirm that https://github.com/eclipse-theia/theia/pull/12295#issuecomment-1472767481 works

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
